core/src/test/java/org/mskcc/cbio/portal/util/TestGenePanelUtil.java

```diff
@@ -0,0 +1,217 @@
+
+        p6 = new Properties();
+        p6.setProperty("stable_id", "TESTPANEL12_SHARED");
+        p6.setProperty("description", "Example gene panel meta file for testing purposes.");
+        p6.setProperty("gene_list", "CADM2\tDTNB\tKAT2A\tMSH3\tMYB\tNPIPB15\tOTOR\tP2RY10");
+    }
+
+    public void setGenes() {
+        g1 = makeGeneSet(otor, cadm2, msh3, p2ry10, adamts20, kat2a, myb, npipb15, dtnb, ablim1);
+        g2 = makeGeneSet(otor, cadm2, piezo1, msh3, fgfr3, p2ry10, kat2a, myb, npipb15, dtnb);
+        g3 = new HashSet<>();
+        g4 = makeGeneSet(adamts20, ablim1);
+        g5 = makeGeneSet(piezo1, fgfr3);
+        g6 = makeGeneSet(otor, cadm2, msh3, p2ry10, kat2a, myb, npipb15, dtnb);
+    }
+
+    @Test
+    public void testExtractPropertyValue() {
+        String expected;
+        String actual;
+
+        actual = extractPropertyValue("stable_id", p1, true);
+        expected = "TESTPANEL1";
+        assertEquals(expected, actual);
+
+        actual = extractPropertyValue("description", p1, false);
+        expected = "Example gene panel meta file for testing purposes.";
+        assertEquals(expected, actual);
+
+        actual = extractPropertyValue("gene_list", p1, true);
+        expected = "ABLIM1\tADAMTS20\tCADM2\tDTNB\tKAT2A\tMSH3\tMYB\tNPIPB15\tOTOR\tP2RY10";
+        assertEquals(expected, actual);
+
+        // gene_list only contains \t as delimiters
+        actual = extractPropertyValue("gene_list", p1, false);
+        assertEquals(expected, actual);
+
+        // cannot test on p3 since its gene_list is empty
+    }
+
+    @Test
+    public void testExtractGenes() {
+        Set<CanonicalGene> expected;
+        Set<CanonicalGene> actual;
+
+        // Check for a non-empty result
+        expected = new HashSet<>();
+        actual = extractGenes(p4, false);
+
+        HashSet<String> hs = makeAliases("ADAM-TS20", "GON-1", "ADAMTS-20");
+        CanonicalGene cg = makeGene(15009, 80070, "ADAMTS20", hs);
+        expected.add(cg);
+
+        hs = makeAliases("LIMAB1", "abLIM-1", "ABLIM", "LIMATIN");
+        cg = makeGene(3239, 3983, "ABLIM1", hs);
+        expected.add(cg);
+
+        assertEquals(expected, actual);
+
+        // Check for an empty result
+        expected = new HashSet<>();
+        actual = extractGenes(p3, true);
+        assertEquals(expected, actual);
+    }
+
+    private HashSet<String> makeAliases(String... aliases) {
+        return new HashSet<>(Arrays.asList(aliases));
+    }
+
+    private HashSet<CanonicalGene> makeGeneSet(CanonicalGene... genes) {
+        return new HashSet<>(Arrays.asList(genes));
+    }
+
+    private CanonicalGene makeGene(int geneticEntityId, long entrezGeneId, String hugoGeneSymbol, Set<String> aliases) {
+        CanonicalGene gene = new CanonicalGene(geneticEntityId, entrezGeneId, hugoGeneSymbol, aliases);
+        gene.setType("protein-coding");
+        return gene;
+    }
+
+    @Test
+    public void testGetAddRemoveAEqualsB() {
```

Comment:
We don't need the duplicate examples for each test case. It just makes this harder to read. It doesn't really add much in terms of additional verification.